CustomEvents.cpp

```diff
@@ -131,6 +139,250 @@ void CustomEventsParser::EarlyParseCustomEventNode(rapidxml::xml_node<char> *nod
+        if (strcmp(creditNode->name(), "scrollPause") == 0)
+        {
+            if (auto pausePositionAttribute = creditNode->first_attribute("pausePosition")) 
+            {
+                try 
+                {
+                    pausePosition = std::stof(pausePositionAttribute->value());
+                } 
+                catch (const std::invalid_argument& e) 
+                {
+                    hs_log_file("Failed to convert an attribute value in 'pausePosition' to a float. Using default value.\n");
+                }
+            }
+
+            if (auto durationAttribute = creditNode->first_attribute("duration")) 
+            {
+                try 
+                {
+                    pauseDuration = std::stof(durationAttribute->value()) * 50 + 250;
+                } 
+                catch (const std::invalid_argument& e) 
+                {
+                    hs_log_file("Failed to convert an attribute value in 'duration' to a float. Using default value.\n");
+                }
+            }
+        }
+
+        // Parse values for credits.txt
+        if (strcmp(creditNode->name(), "creditNames") == 0)
+        {   
+            if (auto valueAttribute = creditNode->first_attribute("fontSize"))
+            {
+                std::string valueStr = valueAttribute->value();
+                try
+                {
+                    creditNamesFontSize = std::stoi(valueStr);
+                }
+                catch (const std::invalid_argument& e)
+                {
+                    hs_log_file("Failed to convert an attribute value in 'creditNames' to an integer. Using default value.\n");
+                }
+            }
+
+            if (auto valueAttribute = creditNode->first_attribute("spacingBetweenNames"))
+            {
+                std::string valueStr = valueAttribute->value();
+                try
+                {
+                    creditNamesLineSpacing = std::stoi(valueStr);
+                }
+                catch (const std::invalid_argument& e)
+                {
+                    hs_log_file("Failed to convert an attribute value in 'creditNames' to an integer. Using default value.\n");
+                }
+            }
+
+            if (auto valueAttribute = creditNode->first_attribute("spacingToFinishTexts"))
+            {
+                std::string valueStr = valueAttribute->value();
+                try
+                {
+                    creditNamesToFinishTextSpacing = std::stoi(valueStr);
+                }
+                catch (const std::invalid_argument& e)
+                {
+                    hs_log_file("Failed to convert an attribute value in 'creditNames' to an integer. Using default value.\n");
+                }
+            }
+        }
+    }
+    
+    // Parse contents in credit.txt
+    CreditFile creditFile;
+    std::string creditsTextFile = G_->GetResources()->LoadFile("data/credits.txt");
+
+    std::stringstream ss(creditsTextFile);
```

Comment:
It looks like every other instance of `LoadFile` in the HS code assigns it to a `char*` instead of an `std::string`. To maintain that pattern I'd like this code rewritten slightly:
```c++
char* creditsTextFile = G_->GetResources()->LoadFile("data/credits.txt");
std::stringstream ss(creditsTextFile);
delete [] creditsTextFile;
```